server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/TextCalendarBodyToAttachmentTest.java

```diff
@@ -0,0 +1,237 @@
+        String messageContent = "Content-type: text/calendar; method=REPLY; charset=UTF-8\n" +
+            "Content-transfer-encoding: 8BIT\n" +
+            "\n" +
+            "BEGIN:VCALENDAR";
+        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(messageContent.getBytes()));
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .build();
+
+        mailet.service(mail);
+
+        assertThat(mail.getMessage().isMimeType("multipart/*")).isTrue();
+    }
+
+    @Test
+    public void serviceShouldConvertTextBodyOfMessageToAttachmentWhenTextCalendar() throws Exception {
+        String messageContent = "Content-type: text/calendar; method=REPLY; charset=UTF-8\n" +
+            "Content-transfer-encoding: 8BIT\n" +
+            "\n" +
+            "BEGIN:VCALENDAR\n" +
+            "END:VEVENT\n" +
+            "END:VCALENDAR";
+        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(messageContent.getBytes()));
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .build();
+
+        mailet.service(mail);
+
+        MimeMessage actual = mail.getMessage();
+        Multipart multipart = (Multipart)actual.getContent();
+
+        assertThat(multipart.getCount()).isEqualTo(1);
+        assertThat(multipart.getBodyPart(0).getDisposition()).isEqualTo("attachment");
+    }
+
+    @Test
+    public void serviceShouldKeepAllMessageHeaderWhenConverting() throws Exception {
+        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(MESSAGE_CONTENT.getBytes()));
+        List<Header> allHeaders = Collections.list(message.getAllHeaders());
+        List<String> allStringHeaders = allHeaders.stream()
+            .map(Header::getName)
+            .collect(Guavate.toImmutableList());
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .build();
+
+        mailet.service(mail);
+
+        List<Header> actualHeaders = Collections.list(mail.getMessage().getAllHeaders());
+        List<String> actualStringHeaders = actualHeaders
+            .stream()
+            .map(Header::getName)
+            .collect(Guavate.toImmutableList());
+
+        assertThat(actualStringHeaders).containsAll(allStringHeaders);
+    }
+
+    @Test
+    public void contentTypeOfAttachmentShouldGetFromOriginalMessage() throws Exception {
+        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(MESSAGE_CONTENT.getBytes()));
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .build();
+
+        mailet.service(mail);
+
+        Multipart multipart = (Multipart)mail.getMessage().getContent();
+
+        int firstBodyPart = 0;
+        assertThat(multipart.getBodyPart(firstBodyPart).getContentType()).startsWith("text/calendar");
+    }
+}
```

Comment:
Show me that you preserved text/calendar content.

With david email, content type is 

```
Content-Type: text/calendar; method=REPLY; charset=UTF-8
```

Is it also the text/calendar part content type after your transformation?